### PR TITLE
refactor installation examples to lazily bootstrap in order to parallelize

### PR DESCRIPTION
**Pull request by jpsim**
Should address #2678 but is blocked on us resolving #2701. /cc @bdash

### CODE REVIEW

File: examples/installation/build.sh [hunk 1]
```diff
@@ -30,152 +28,121 @@ EOF
...
-xctest_osx() {
-    XCODE_COMMAND="$@"
-    xcodebuild $XCODE_COMMAND clean build test -sdk macosx || exit 1
+xctest() {
+    PLATFORM="$1"
+    LANG="$2"
+    NAME="$3"
+    DIRECTORY="$PLATFORM/$LANG/$NAME"
+    PROJ="$DIRECTORY/$NAME.xcodeproj"
+    WORK="$DIRECTORY/$NAME.xcworkspace"
```
Comment (bdash) [inline examples/installation/build.sh]: `PROJECT` and `WORKSPACE` would be fine.

**Comment by bdash** [comment]
Seems fine. Do you expect this to have issues on CI?

**Comment by jpsim** [comment]
We still need to create a matrix job on CI for all of these to run in parallel. Other than that and #2706, I don't expect this to cause issues.

**Comment by jpsim** [comment]
CI failures unrelated, filed a radar for the `xcodebuild` crash: http://www.openradar.me/23172400

**Comment by bdash** [comment]
:+1:

It'd be good to confirm that the `xcodebuild` crash is transient and not a bug that this refactoring brings to light.

**Comment by jpsim** [comment]
> It'd be good to confirm that the xcodebuild crash is transient and not a bug that this refactoring brings to light.

I can confirm that because the only file modified by this PR is never referenced in CI's PR jobs. Only in releasability.

**Pull request merged**